core/src/main/java/org/apache/calcite/plan/MaterializedViewSubstitutionVisitor.java

```diff
@@ -259,6 +263,77 @@ public UnifyResult apply(UnifyRuleCall call) {
     }
   }
 
+  /**
+   * Implementation of {@link SubstitutionVisitor.UnifyRule} that matches a
+   * {@link MutableProject} on top of a {@link MutableAggregate} to a
+   * {@link MutableProject} on top of a {@link MutableAggregate}.
+   *
+   * <p>Example: aggregate calls of query is a subset of target.</p>
+   * <ul>
+   * <li>query:   Project(projects: [$0, *(2, $1)])
+   *                Aggregate(groupSet: {0}, groupSets: [{0}], calls: [SUM($1)])
+   *                  Scan(table: [hr, emps])</li>
+   * <li>target:  Project(projects: [$0, *(2, $1), *(2, $2)])
+   *                Aggregate(groupSet: {0}, groupSets: [{0}], calls: [SUM($1), COUNT()])
+   *                  Scan(table: [hr, emps])</li>
+   * </ul>
+   */
+  private static class ProjectOnAggregateToProjectOnAggregateUnifyRule
```

Comment:
Maybe you can rename the rule name to `ProjectAggregateToProjectAggregateUnifyRule `

Reply:
THX for look into this @hsyuan 
I will update soon.